wgengine/magicsock/magicsock.go

```diff
@@ -2533,6 +2533,29 @@ func (c *Conn) SetDERPMap(dm *tailcfg.DERPMap) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	var derpAddr = debugUseDERPAddr()
+	if derpAddr != "" {
```

Comment:
May want to consider putting the assignment in the if statement, that way `derpAddr` is scoped to it and is easier to reason about (the reader doesn't have to wonder if it's used later in the function).

```suggestion
	if derpAddr := debugUseDERPAddr(); derpAddr != "" {
```